guix: Update darwin native packages dependencies

hebasto opened the pull request:
It is a #20470 follow up.

hebasto committed:
guix: Update darwin native packages dependencies

hebasto commented:
Guix build:

```
$ find output -type f -name *$(git rev-parse --short HEAD)*.* -print0 | sort -z | xargs -r0 sha256sum
c39d6081b90a28eb477ecff074d803888adb163eaad914d1ac767e96f8aa713d  output/bitcoin-d8b4e3b43832-osx64.tar.gz
1040e7b19e9fd482760e23d1fda199c89f96a77f3a07dae009b355c3412d0b64  output/bitcoin-d8b4e3b43832-osx-unsigned.dmg
ea29f074ba4a6b9707ceda4b8c2c1349f048788cfb06c44e72411ac7ae3f1907  output/bitcoin-d8b4e3b43832-osx-unsigned.tar.gz
09189d3408f3b313036214822142223f2e992006d619ba48bfaec8e907ab16ed  output/src/bitcoin-d8b4e3b43832.tar.gz
```

practicalswift commented:
cr ACK d8b4e3b43832beb4221974aecc93de96b8a0e9e4: patch looks correct

fanquake commented on contrib/guix/libexec/build.sh:
```diff
@@ -61,9 +61,10 @@ case "$HOST" in
-        for native_pkg in libcap zlib bzip2; do
+        # zlib         required by native_libdmg-hfsplus
+        #
+        # shellcheck disable=SC2043
```
The change looks correct, but can't we just remove the loop, rather than adding more shellcheck cruft to disable a warning about what is a pointless loop? You could then also make the comment above zlib specific. i.e: "When targeting darwin, a native zlib is required  by libdmg-hfsplus", and just remove the listing.

hebasto replied:
My thought was if another dependency appears in the future this approach allows to just add a package name into the loop.

But your suggestion is more neat. Going to implement it.

hebasto replied:
Thanks! [Updated](https://github.com/bitcoin/bitcoin/pull/21337#issuecomment-789484190).

fanquake replied:
> My thought was if another dependency appears in the future this approach allows to just add a package name into the loop.

If we need to add more packages (I'd think unlikely at this stage), we can just re-add the loop.

fanquake reviewed (commented):
Concept ACK

hebasto commented:
Updated d8b4e3b43832beb4221974aecc93de96b8a0e9e4 -> 7bbb409314c63750a9731dedaaf6dd6ecd17d988 ([pr21337.01](https://github.com/hebasto/bitcoin/commits/pr21337.01) -> [pr21337.02](https://github.com/hebasto/bitcoin/commits/pr21337.02), [diff](https://github.com/hebasto/bitcoin/compare/pr21337.01..pr21337.02)):

- addressed @fanquake's [comment](https://github.com/bitcoin/bitcoin/pull/21337#discussion_r586050480)

hebasto commented:
Guix build:
```
$ find output -type f -name *$(git rev-parse --short HEAD)*.* -print0 | sort -z | xargs -r0 sha256sum
c2a7960e19f06ed2f78b50e88c099956e590c4559026f2f68e92d1d7b28d5eba  output/bitcoin-7bbb409314c6-osx64.tar.gz
ff897bb2b361a3e6e8d9a96add8010061b09bb5c0f880a7e86dd716da18d8f4a  output/bitcoin-7bbb409314c6-osx-unsigned.dmg
1b60543e7b9e5e4ea40cf4e454cea1d7570e7b814760b142a7f0f4de428e96d8  output/bitcoin-7bbb409314c6-osx-unsigned.tar.gz
fb01097e36dc0b594c267bbd7ed222a24ce34f98c32be533eaaa8aab462189b1  output/src/bitcoin-7bbb409314c6.tar.gz
```

fanquake reviewed (approved):
ACK 7bbb409314c63750a9731dedaaf6dd6ecd17d988 - thanks for noticing the leftovers. This looks ok to me. Guix builds look like they are matching:

```bash
bash-5.1# find output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum
ff897bb2b361a3e6e8d9a96add8010061b09bb5c0f880a7e86dd716da18d8f4a  output/bitcoin-7bbb409314c6-osx-unsigned.dmg
1b60543e7b9e5e4ea40cf4e454cea1d7570e7b814760b142a7f0f4de428e96d8  output/bitcoin-7bbb409314c6-osx-unsigned.tar.gz
c2a7960e19f06ed2f78b50e88c099956e590c4559026f2f68e92d1d7b28d5eba  output/bitcoin-7bbb409314c6-osx64.tar.gz
fb01097e36dc0b594c267bbd7ed222a24ce34f98c32be533eaaa8aab462189b1  output/src/bitcoin-7bbb409314c6.tar.gz
```

dongcarl commented:
Concept ACK 7bbb409314c63750a9731dedaaf6dd6ecd17d988

Lovely! ☺️ 
I believe we can also remove `libcap` from the manifest?

hebasto committed:
guix: Remove libcap from manifest

hebasto commented:
Updated 7bbb409314c63750a9731dedaaf6dd6ecd17d988 -> c967fb7fb97a86fc434912d0fff621f9f50dfe80 ([pr21337.02](https://github.com/hebasto/bitcoin/commits/pr21337.02) -> [pr21337.03](https://github.com/hebasto/bitcoin/commits/pr21337.03), [diff](https://github.com/hebasto/bitcoin/compare/pr21337.02..pr21337.03)):

- addressed @dongcarl's [comment](https://github.com/bitcoin/bitcoin/pull/21337#issuecomment-789998344):
> I believe we can also remove `libcap` from the manifest?

fanquake reviewed (approved):
ACK c967fb7fb97a86fc434912d0fff621f9f50dfe80

Guix builds at c967fb7fb97a86fc434912d0fff621f9f50dfe80:
```bash
bash-5.1# find output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum
2cece5e8cc7c5e97ea9691788b02fcc2b7b5a791db12de8e0ef0655ecbfe694d  output/bitcoin-c967fb7fb97a-osx-unsigned.dmg
b3c4f71a656ac1873a12b4a080d2cbc93b7e221ea298ebdb393b397c2c36d76e  output/bitcoin-c967fb7fb97a-osx-unsigned.tar.gz
b0244e16d42f606f91de9dc455c33b01341f03733a0c48c9b2db27a98d02afb7  output/bitcoin-c967fb7fb97a-osx64.tar.gz
2f8d12a78e03f9a3d2971fb9bc8ff9794587557b02a6773992df860041e989d3  output/src/bitcoin-c967fb7fb97a.tar.gz
```

laanwj commented:
```
$ find output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum
2cece5e8cc7c5e97ea9691788b02fcc2b7b5a791db12de8e0ef0655ecbfe694d  output/bitcoin-c967fb7fb97a-osx-unsigned.dmg
56c4885e2fd20b09f2acd455ec4c6644a5afc9e7f51803933894b3ed77ed68bc  output/bitcoin-c967fb7fb97a-osx-unsigned.tar.gz
b0244e16d42f606f91de9dc455c33b01341f03733a0c48c9b2db27a98d02afb7  output/bitcoin-c967fb7fb97a-osx64.tar.gz
2f8d12a78e03f9a3d2971fb9bc8ff9794587557b02a6773992df860041e989d3  output/src/bitcoin-c967fb7fb97a.tar.gz
```

It looks like my `output/bitcoin-c967fb7fb97a-osx-unsigned.tar.gz` is different (…d68bc instead of …6d76e). In case anyone wants to compare it can be found [here](https://download.visucore.com/tmp/bitcoin-c967fb7fb97a-osx-unsigned.tar.gz). The others match.

fanquake commented:
@laanwj that same difference is also [popping up elsewhere](https://github.com/bitcoin/bitcoin/pull/20980#issuecomment-790270349), and iirc is related to the last Guix update. It should be fixed in the one of the next Guix PRs, but shouldn't block changes here.

laanwj commented:
Ok, good to know, will continue to merge this then. I just hoped posting the output would be useful to resolve determinism issues.
